crates/evm/src/evm/handler.rs

```diff
@@ -314,6 +314,29 @@ struct CitreaHandler<SPEC, EXT, DB> {
 }
 
 impl<SPEC: Spec, EXT: CitreaExternalExt, DB: Database> CitreaHandler<SPEC, EXT, DB> {
+    fn load_precompiles() -> ContextPrecompiles<DB> {
+        fn our_precompiles<SPEC: Spec, DB: Database>() -> ContextPrecompiles<DB> {
+            use revm::precompile::{
+                u64_to_address, Bytes, Precompile, PrecompileOutput, PrecompileResult,
+            };
+
+            pub fn kzg(_input: &Bytes, _gas_limit: u64, _env: &Env) -> PrecompileResult {
+                pub const GAS_COST: u64 = 0;
+                pub const RETURN_VALUE: &[u8; 64] = &[0; 64];
+                Ok(PrecompileOutput::new(GAS_COST, RETURN_VALUE.into()))
+            }
+
+            let mut precompiles = revm::handler::mainnet::load_precompiles::<SPEC, DB>();
+            let precompiles_inner = precompiles.to_mut();
+            if let Some(kzg_point_evaluation) = precompiles_inner.get_mut(&u64_to_address(0x0A)) {
```

Comment:
since we know at what point kzg comes into play, we can remove only then with the index :)

Reply:
also the costs should not be zero. it should be the same as call gas cost to a regular address with no precompile